src/nemos/utils.py

```diff
@@ -123,21 +123,21 @@ def convolve_1d_trials(
 
     # Check window size
     ws = len(basis_matrix)
-    if any(trial.shape[0] < ws for trial in time_series):
+    if pytree_map_and_reduce(lambda x: x.shape[0] < ws, any, list(time_series)):
         raise ValueError(
             "Insufficient trial duration. The number of time points in each trial must "
             "be greater or equal to the window size."
         )
 
     if isinstance(time_series, jnp.ndarray):
         # if the conversion to array went through, time_series have trials with equal size
-        conv_trials = list(_CORR_SAME_TRIAL_DUR(time_series, basis_matrix))
+        conv_trials = _CORR_SAME_TRIAL_DUR(time_series, basis_matrix)
     else:
         # trials have different length
-        conv_trials = [
-            _CORR_VARIABLE_TRIAL_DUR(jnp.atleast_2d(trial), basis_matrix)
-            for trial in time_series
-        ]
+        conv_trials = jax.tree_map(
+            lambda x: _CORR_VARIABLE_TRIAL_DUR(jnp.atleast_2d(x), basis_matrix),
+            time_series, basis_matrix
```

Comment:
I think this is a typo right? Putting `basis_matrix` as an arg to tree_map means that it should be a pytree with the same structure as `time_series`, but including in the call the `_CORR_VARIABLE_TRIAL_DUR` means that it's being correlated with each leaf. shouldn't it be something like:
```suggestion
            lambda x, y: _CORR_VARIABLE_TRIAL_DUR(jnp.atleast_2d(x), y),
            time_series, basis_matrix
```

Reply:
Or it should be:
```suggestion
            lambda x: _CORR_VARIABLE_TRIAL_DUR(jnp.atleast_2d(x), basis_matrix),
            time_series
```